src/eterna/puzzle/PuzzleManager.ts

```diff
@@ -81,7 +82,8 @@ export default class PuzzleManager {
     }
 
     public async parsePuzzle(json: PuzzleJSON): Promise<Puzzle> {
-        const newpuz: Puzzle = new Puzzle(Number(json['id']), json['title'], json['type'], json['username']);
+        const newpuz: Puzzle = new Puzzle(Number(json['id']),
```

Comment:
Please follow the current convention for optional data, which is like
```
if (json['whatever']) newpuz.whatever = json['whatever']
```
instead of including it as a constructor parameter